source/js/components/takeover/takeover.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+export default class Takeover extends React.Component {
+  show () {
+    this.setState({
+      isHidden: false
+    });
+  }
+
+  hide () {
+    window.scrollTo(0, 0);
+
+    this.setState({
+      isHidden: true
+    });
+  }
+
+  toggle () {
+    this.setState({
+      isHidden: !this.state.isHidden
+    });
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.hide = this.hide.bind(this);
```

Comment:
To do that I believe we have to refactor our build system. If we do that then we can remove all the bindings as part of that...